src/Writer/Factory/WriterFactory.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Log\Writer\Factory;
+
+use Interop\Container\ContainerInterface;
+use Zend\ServiceManager\Exception\InvalidServiceException;
+use Zend\ServiceManager\FactoryInterface;
+use Zend\ServiceManager\ServiceLocatorInterface;
+
+/**
+ * Factory for instantiating classes with no dependencies or which accept a single array.
+ *
+ * The WriterFactory can be used for any class that:
+ *
+ * - has no constructor arguments;
+ * - accepts a single array of arguments via the constructor.
+ *
+ * It replaces the "invokables" and "invokable class" functionality of the v2
+ * service manager, and can also be used in v2 code for forwards compatibility
+ * with v3.
+ */
+final class WriterFactory implements FactoryInterface
+{
+    /**
+     * Options to pass to the constructor (when used in v2), if any.
+     *
+     * @param null|array
+     */
+    private $creationOptions;
+
+    /**
+     * @param array $creationOptions
+     */
+    public function __construct(array $creationOptions = null)
+    {
+        if (is_array($creationOptions)) {
+            $this->setCreationOptions($creationOptions);
+        }
+    }
+
+    /**
+     * Create an instance of the requested class name.
+     *
+     * @param ContainerInterface $container
+     * @param string $requestedName
+     * @param null|array $options
+     * @return object
+     */
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
+    {
+        $options = (array)$options;
+
+        if (isset($options['filter_manager']) && is_string($options['filter_manager'])) {
+            $options['filter_manager'] = $container->get($options['filter_manager']);
+        } elseif (!isset($options['filter_manager']) && $container->has('LogFilterManager')) {
+            $options['filter_manager'] = $container->get('LogFilterManager');
+        }
+
+        if (isset($options['formatter_manager']) && is_string($options['formatter_manager'])) {
+            $options['formatter_manager'] = $container->get($options['formatter_manager']);
+        } elseif (!isset($options['formatter_manager']) && $container->has('LogFormatterManager')) {
+            $options['formatter_manager'] = $container->get('LogFormatterManager');
+        }
+
+        return new $requestedName($options);
```

Comment:
@weierophinney I've changed this line as well since I'm now casting $options to an array. The old line was `return (null === $options) ? new $requestedName : new $requestedName($options);`

Do you agree?